src/GridEngine.ts

```diff
@@ -795,6 +800,29 @@ export class GridEngine {
     );
   }
 
+  /**
+   * @returns Observable that emits when a new character is added or an existing is removed.
+   */
+  characterShifted(): Observable<{
+    charId: string;
+    action: "REMOVED" | "ADDED";
+  }> {
+    return this.charAdded$.pipe(
+      map<string, { charId: string; action: "REMOVED" | "ADDED" }>((c) => ({
+        charId: c,
+        action: "ADDED",
+      })),
+      mergeWith(
+        this.charRemoved$.pipe(
+          map<string, { charId: string; action: "REMOVED" | "ADDED" }>((c) => ({
+            charId: c,
+            action: "REMOVED",
+          }))
+        )
+      )
+    );
+  }
```

Comment:
How about introducing an enum and an interface? That makes it more readable and in case of a change you don't need to touch 3 type declarations:

```suggestion

/** Some docs here */
  export interface CharacterShift {
    charId: string;
    action: CharacterShiftAction;
  }

/** Some docs here */
  export enum CharacterShiftAction {
    REMOVED = "REMOVED",
    ADDED = "ADDED",
  }

  /**
   * @returns Observable that emits when a new character is added or an existing is removed.
   */
  characterShifted(): Observable<CharacterShift> {
    return this.charAdded$.pipe(
      map((charId) => ({
        charId,
        action: CharacterShiftAction.ADDED,
      })),
      mergeWith(
        this.charRemoved$.pipe(
          map((charId) => ({
            charId,
            action: CharacterShiftAction.REMOVED,
          }))
        )
      )
    );
  }
```